### ISSUE TITLE
AsciiDoc report generation support

### ISSUE THREAD

**Issue by HexHash64** [MooseDojo/Serpico#52]
Ruby has a native AsciiDoctor library, which is a pure-ruby open-source implementation of AsciiDoc.
http://asciidoctor.org/

Instead of relying on .docx formats, by exporting findings via AsciiDoctor, the generated AsciiDoc output can easily be converted to a variety of formats. This includes exporting to HTML5, Docbook, PDF (via asciidoctor-pdf), or even to a pure AsciiDoc format which itself can be exported to many other formats via the binary AsciiDoc program (including exporting to ODF): https://github.com/dagwieers/asciidoc-odf

Adding AsciiDoc as a supported output would make reporting more flexible.

**Comment by BuffaloWill** [comment]
Thank you for the submission. This is an interesting idea. 

We could easily add in support for findings to be exported into an AsciiDoc format. What I have in mind is the something similar to the status report but instead of in docx, the findings are printed to AsciiDoc.  Would this be valuable? Granted, it falls far short of producing a report.

Getting a report into AsciiDoc would be a more complicated process. Because the reports begin as a docx we need to go from docx to AsciiDoc and then insert the findings. Have you seen any document on converting a docx to AsciiDoc? If so, we can follow that convention.

**Comment by HexHash64** [comment]
Yes, I at least getting the findings into AsciiDoc would be very valuable. That would then allow people like me to insert it into our own AsciiDoc based report templates.

As for converting from Docx to AsciiDoc, I've read it is possible using the open-source pandoc tool. However, the process appears to involve converting Docx -> HTML5 -> AsciiDoc.

Alternatively, this website (http://thepaulrayner.com/blog/2013/02/14/word-to-asciidoc) mentions saving the Docx as HTML (presumably within Word) and then using pandoc to convert the HTML to AsciiDoc. Of course, depending on the complexity of the template, the resulting AsciiDoc would likely require tweaking to get the desired appearance.

**Comment by BuffaloWill** [comment]
@HexHash64 : I just pushed the feature for the report findings to AsciiDoc. To use, go to [Report] > Additional Features > Generate AsciiDoc of Current Findings
Does the formating work? I took some liberties with header sizes, etc. 

Thinking about the AsciiDoc report generation more, one option would be allowing the user to upload an AsciiDoc report template (or ship with a simple one) and inserting the findings into it. This would skip having to convert docx to AsciiDoc while still supporting report generation in AsciiDoc.
